File: datasets/lua/HumanEval_26_remove_duplicates.lua
-- From a list of integers, remove all elements that occur more than once.
-- Keep order of elements left the same as in the input.
-- >>> remove_duplicates([1, 2, 3, 2, 4])
-- [1, 3, 4]
local function remove_duplicates(numbers)
  local result = {}
  for _, number in ipairs(numbers) do
    if not result[number] then
      result[number] = true
      table.insert(result, number)
    end
  end
  return result
end
-- Unit tests below

lu = require('luaunit')

function test_humaneval()
local candidate = remove_duplicates
    lu.assertEquals(candidate({}), {})
    lu.assertEquals(candidate({1, 2, 3, 4}), {1, 2, 3, 4})
    lu.assertEquals(candidate({1, 2, 3, 2, 4, 3, 5}), {1, 4, 5})
end

os.exit(lu.LuaUnit.run())

File: datasets/rs/HumanEval_26_remove_duplicates.rs
/// From a list of integers, remove all elements that occur more than once.
/// Keep order of elements left the same as in the input.
/// >>> remove_duplicates([1, 2, 3, 2, 4])
/// [1, 3, 4]
fn remove_duplicates(numbers: Vec<isize>) -> Vec<isize> {
    let mut result: Vec<isize> = Vec::new();
    let mut seen: HashSet<isize> = HashSet::new();
    for number in numbers {
        if !seen.contains(&number) {
            result.push(number);
            seen.insert(number);
        }
    }
    result

}

fn main() {
    let candidate = remove_duplicates;
    assert_eq!(candidate(Vec::<isize>::new()), Vec::<isize>::new());
    assert_eq!(candidate(vec![1, 2, 3, 4]), vec![1, 2, 3, 4]);
    assert_eq!(candidate(vec![1, 2, 3, 2, 4, 3, 5]), vec![1, 4, 5]);
}
